documentation/tutorial.md

```diff
@@ -160,92 +203,100 @@ selectors in Crawly shell.
 
 ```
-{:ok, document} = Floki.parse_document(response.body)
-document |> Floki.find("section.wrapper") |> Floki.find("div.article-tiles.article-tiles--wide a") |> Floki.attribute("href")
+response = Crawly.fetch("https://books.toscrape.com/catalogue/category/books/travel_2/index.html")
 
-["/our-range/tools/power-tools", "/our-range/tools/garage-storage",
- "/our-range/tools/hand-tools", "/our-range/tools/tool-storage",
- "/our-range/tools/ladders", "/our-range/tools/safety-equipment-and-workwear",
- "/our-range/tools/work-benches"]
 ```
 
-The result of running the command above is a list of elements which
-contain href attribute of links selected with
-`a.category-block-heading__title` CSS selector. These URLs will be
-used in order to feed Crawly with requests to follow.
-
-In order to find the proper CSS selectors to use, you might find
-useful opening the target page from the shell in your web browser. You
-can use your browser developer tools to inspect the HTML and come up
-with a selector.
-
-Now let's navigate to one of the Homebase product pages and extract
-data from it.
+All books on the page are enclosed in a `.product_pod` class. We can the select the class and pick the information we want from it.
 
 ```
-response = Crawly.fetch("https://www.homebase.co.uk/bosch-universalimpact-800-corded-hammer-drill_p494894")
-
+items =
+  document
+  |> Floki.find(".product_pod")
+  |> Enum.map(fn x ->
+    %{
+      title: Floki.find(x, "h3 a") |> Floki.attribute("title"),
+      price: Floki.find(x, ".product_price .price_color") |> Floki.text()
+    }
+  end)
 ```
 
-Extract the `title` with:
+Here we select all products using the class name, iterate through it and pick the title using `Floki.find(x, "h3 a") |> Floki.attribute("title")`
+and pick the price using `Floki.find(x, ".product_price .price_color") |> Floki.text()`. The full text of the book name is only contained
+in the title attribute and not in the text node of the tag, hence our use of the title attribute.
+
+Also, the data is paginated and we are not only interested in the first page, we have to tell crawly to also visit the next page
+by creating a request struct with the url of the next page and other necessary data.
 
 ```
-{:ok, document} = Floki.parse_document(response.body)
-document |> Floki.find(".page-title h1") |> Floki.text()
-"Bosch UniversalImpact 800 Corded Hammer Drill"
+next_url =
+  document 
+  |> Floki.find(".next a") 
+  |> Floki.attribute("href")
+  |> Floki.text()
+
+next_request =
+  case next_url do
+    "" ->
+      []
+
+    url ->
+      new_request =
+        build_absolute_url(response.request.url, url)
+        |> Crawly.Utils.request_from_url()
+
+      [new_request]
+  end
 ```
 
-Extract the `SKU` with:
+Here we get the url for the next page using Floki, which could be an empty string if such doesn't exist. We then create a request struct
+by piping the url we got from Floki to `build_absolute_url/2` and `Crawly.Utils.request_from_url`.
```

Comment:
update to explain that if it doesn't exist (e.g. end of page) , it will simply return an empty list, and result in spider stopping.

Don't need to specially handle situation where only one exists etc.

Reply:
Done